modules/table/src/integrationTest/java/org/apache/ignite/internal/rebalance/ItRebalanceDistributedTest.java

```diff
@@ -660,6 +663,63 @@ void testRaftClientsUpdatesAfterRebalance() throws Exception {
 
     }
 
+    @Test
+    void testClientsAreUpdatedAfterPendingRebalanceHandled() throws Exception {
+        Node node = getNode(0);
+
+        createZone(node, ZONE_NAME, 1, 1);
+
+        createTable(node, ZONE_NAME, TABLE_NAME);
+
+        assertTrue(waitForCondition(() -> getPartitionClusterNodes(node, 0).size() == 1, AWAIT_TIMEOUT_MILLIS));
+
+        String assignmentsBeforeRebalance = getPartitionClusterNodes(node, 0).stream()
+                .findFirst()
+                .orElseThrow()
+                .consistentId();
+
+        String newNodeNameForAssignment = nodes.stream()
+                .filter(n -> !assignmentsBeforeRebalance.equals(n.clusterService.nodeName()))
+                .findFirst()
+                .orElseThrow()
+                .name;
+
+        Set<Assignment> newAssignment = Set.of(Assignment.forPeer(newNodeNameForAssignment));
+
+        // Write the new assignments to metastore as a pending assignments.
+        TablePartitionId partId = new TablePartitionId(getTableId(node, TABLE_NAME), 0);
+
+        ByteArray partAssignmentsPendingKey = pendingPartAssignmentsKey(partId);
+
+        byte[] bytesPendingAssignments = ByteUtils.toBytes(newAssignment);
+
+        AtomicBoolean stopDropping = new AtomicBoolean(true);
+
+        // Using this hack we pause rebalance on all nodes
+        nodes.forEach(n -> ((DefaultMessagingService) n.clusterService.messagingService())
+                .dropMessages((nodeName, msg) -> msg instanceof ChangePeersAsyncRequest && stopDropping.get())
+        );
+
+        node.metaStorageManager.put(partAssignmentsPendingKey, bytesPendingAssignments).get(AWAIT_TIMEOUT_MILLIS, MILLISECONDS);
+
+        // Check that raft clients on all nodes were updated with the new list of peers.
+        assertTrue(waitForCondition(
+                () -> nodes.stream().allMatch(n ->
+                        n.tableManager
+                                .latestTables()
+                                .get(getTableId(node, TABLE_NAME))
+                                .internalTable()
+                                .partitionRaftGroupService(0)
+                                .peers()
+                                .stream()
+                                .collect(toSet())
+                                .equals(Set.of(new Peer(newNodeNameForAssignment), new Peer(assignmentsBeforeRebalance)))),
+                (long) AWAIT_TIMEOUT_MILLIS * nodes.size()
+        ));
+
+        stopDropping.set(false);
```

Comment:
And what about verification that will check that there are only stable assignments in client.peers() on rebalance finished?

Reply:
well, I could add it, but in general, such logic is checked in `ItRebalanceDistributedTest#testRaftClientsUpdatesAfterRebalance`

Reply:
No need in that case.